server/src/main/java/org/apache/iotdb/db/query/control/QueryResourceManager.java

```diff
@@ -90,6 +90,19 @@ public long assignQueryId(boolean isDataQuery) {
     return queryId;
   }
 
+  public long assignCompactionQueryId() {
+    long threadNum;
+    try {
+      threadNum = Long.parseLong((Thread.currentThread().getName().split("-"))[3]);
+    } catch (ArrayIndexOutOfBoundsException e) {
+      // for test
+      threadNum = 0;
+    }
+    long queryId = Long.MIN_VALUE + threadNum;
+    filePathsManager.addQueryId(queryId);
+    return queryId;
+  }
```

Comment:
Better add some docs about the generation of the special query id for compaction, and avoid others use the same way to generate specail query id for other usage.

Reply:
resolved